File: S05/prob13.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jul 20 15:20:54 2022

@author: Gerardo A. Rivera Tello
@handle: DangoMelon
"""

while (num := eval(input("Ingrese un numero: "))) < 0:
    continue

fact = 1

for i in range(1, num + 1):
    fact *= i

print(f"El factorial de {num} es: {fact}")

File: S05/prob13.cpp
/*
 * Created on Wed Jul 20 15:23:24 2022

 * @author: Gerardo A. Rivera Tello
 * @handle: DangoMelon
 */

#include <iostream>

int main()
{
    int num;
    unsigned long long int factorial = 1;
    do
    {
        std::cout << "Ingrese un numero: ";
        std::cin >> num;
    } while (num < 0);

    for (int i = 1; i <= num; i++)
    {
        factorial *= i;
    }
    std::cout << "El factorial de " << num << " es: " << factorial << std::endl;
}